### PR TITLE
Add configuration

### PR DESCRIPTION
**Pull request by thephw**
### Changelog
- Brings back kustomize templates as a prerequisite for the CI suite
- Update lifecycle scripts to accept a kustomize overlay
- Add development overlay to gitignore
- Add code of conduct
- Add LICENSE

### CODE REVIEW

**Comment by sj26** [comment]
There's a lot going on in here! I'd love to see some smaller, more focused PRs.

**Comment by thephw** [comment]
This has been re-focused for only the configuration changes and rebased to the current main branch.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 Brewfile.lock.json
+
+k8s/overlays/thephw/
```
Comment (sj26) [inline .gitignore:3]:
```suggestion
k8s/overlays/*
-k8s/overlays/example
```

File: README.md [hunk 2]
```diff
@@ -6,56 +6,90 @@ We've seen many customers running the Agent on their own Kubernetes clusters. Th
...
    2. SSH key
 
-Then apply the manifest to your cluster. For example:
-
 ```
-kubectl apply -f k8s/buildkite.yaml
+mkdir k8s/overlays/my-stackname
+touch k8s/overlays/my-stackname/git-credentials
+touch k8s/overlays/my-stackname/private-ssh-key
+touch k8s/overlays/my-stackname/kustomization.yaml
```
Comment (sj26) [inline README.md:19]: cp -r k8s/overlays/example k8s/overlays/my-stackname

File: README.md [hunk 3]
```diff
@@ -6,56 +6,90 @@ We've seen many customers running the Agent on their own Kubernetes clusters. Th
...
 
 ```
-printf "<PASTE_BUILDKITE_AGENT_TOKEN>" | base64
+k8s/overlays/my-stackname
 ```
 
+### Agent token
+
+The buildkite agent token can be found here for your GitHub organization.
+https://buildkite.com/organizations/MY_ORGANIZATION_SLUG/agents
```
Comment (sj26) [inline README.md:55]:
```suggestion
https://buildkite.com/organizations/~/agents
```

File: README.md [hunk 4]
```diff
@@ -6,56 +6,90 @@ We've seen many customers running the Agent on their own Kubernetes clusters. Th
...
+Then you should either make your repository clone private OR add your overlay to your `.gitignore` file. These secrets should not be checked into a public repository.
 
 ```
-printf "<PASTE_BUILDKITE_AGENT_TOKEN>" | base64
+k8s/overlays/my-stackname
 ```
 
+### Agent token
+
+The buildkite agent token can be found here for your GitHub organization.
```
Comment (sj26) [inline README.md:54]:
```suggestion
Your Buildkite agent token can be found here:
```

File: README.md [hunk 5]
```diff
@@ -6,56 +6,90 @@ We've seen many customers running the Agent on their own Kubernetes clusters. Th
...
+https://buildkite.com/organizations/MY_ORGANIZATION_SLUG/agents
+
+Paste that value into the space labeled "PASTE_AGENT_TOKEN_HERE" in `/k8s/overlays/my-stackname`
```
Comment (sj26) [inline README.md:57]:
```suggestion
Paste that value into the space labeled "PASTE_AGENT_TOKEN_HERE" in `k8s/overlays/my-stackname/kustomization.yaml`
```

File: README.md [hunk 6]
```diff
@@ -6,56 +6,90 @@ We've seen many customers running the Agent on their own Kubernetes clusters. Th
...
 #### Git credentials
 
 You can create a set of git credentials for testing on GitHub [here](https://github.com/settings/tokens). You only need to select repository access.
 
-To encode and copy the git credentials run the below and paste the result into `buildkite.yaml` where it reads `PASTE_GIT_CREDENTIALS_HERE`
+Create the git credentials file by running the command below with the template values filled in:
 
 ```
-printf "https://<MY_GITHUB_USERNAME>:<MY_ACCESS_TOKEN>@github.com" | base64
+echo "https://<MY_GITHUB_USERNAME>:<MY_ACCESS_TOKEN>@github.com" > ./k8s/overlays/my-stackname/git-credentials
```
Comment (sj26) [inline README.md:68]: I reckon go simpler:

```
Edit `k8s/overlays/my-stackname/git-credentials` and add a line like:

    https://<MY_GITHUB_USERNAME>:<MY_ACCESS_TOKEN>@github.com
```

**Review by sj26 — APPROVED** [review]
Nice, love the general shape of this.

I reckon we should gitignore everything in the overlays directory _except_ I think we should provide an example overlay and suggest copying it. I've made some suggestions which roughly fill out that idea.

File: .gitignore (hunk 1 above)
Comment (thephw) [reply .gitignore]: Updated.

File: README.md (hunk 2 above)
Comment (thephw) [reply README.md]: Updated.

**Pull request merged**